### DISCUSSION TITLE
Years in sidebar missing month links

### DISCUSSION THREAD

**Discussion by MaffooClock** [Daveiano/weewx-wdc#285] [Q&A] [answered]
I've been running a weewx weather station for about 14 years (not long after weewx was born).  My weewx server and website have been neglected for the past couple of years, so I put in some work to get everything up-to-date, and even implemented weewx-wdc for a fresh new experience.

Setting up weewx-wdc took a lot of tinkering to get it the way I wanted it.  Now that I'm done, I've noticed a problem that I haven't been able to figure out (yet): in the side navigation along the left side, beneath each year listed, there should be links to each month for that year (the `month-%Y-%m.html`page)... but they're missing, although the `month-%Y-%m.html` pages do exist.

The Years item on every page does list all of the years, though.  And on every `year-%Y.html` and `month-%Y-%m.html` page, the missing month links do appear.  So those month links are only missing on these pages:
- `index.html`
- `yesterday.html`
- `week.html`
- `month.html`
- `year.html`
- `celestial.html`
- `statistics.html`

[See for yourself](https://wx.kd5hia.net/).

I'm pretty sure the `month-%Y-%m.html` links were working before, so I don't know what happened that caused them not be present during report generation.  And earlier when I said "a lot of tinkering," I just meant tweaking the `skin.conf` -- I didn't make any changes to the skin itself.

I have run `weectl database drop-daily` and `weectl database rebuild-daily`, as well as clearing all files from `HTML_ROOT` before manually regenerating the entire site via `sudo -u weewx weectl report run WdcReport`.  There are no errors when the report generator is running.

My next step is to review the source of the skin to learn how the templates are generated so I can understand the logic that determines whether to output a menu item, but that'll take a little time and effort.  In the meantime, I figured I'd ask about this in case someone else with deeper knowledge of this skin might be able to share some direction.

**Comment by MaffooClock** [comment]
I found where the month links are generated:
https://github.com/Daveiano/weewx-wdc/blob/8006446396fa5c2f0eaf69608d35bb9f991451f1/skins/weewx-wdc/includes/ui-shell.inc#L350-L362

...and it appears that `$archive_months` (which is actually `$SummaryByMonth`) is empty on certain pages (listed in initial post).

So now we know why the links aren't being generated, now to find out why that variable is empty on some pages but not others...

**Comment by MaffooClock** [comment, marked as answer]
Okay, figured it out: it's because I'm stupid.

Early on in the implementation of weewx-wdc, I moved the `[[ToDate]]` section above the `[[SummaryBy____]]` sections so that I could play with the Today page without having to wait an hour for all the `month-%Y-%m.html` and `year-%Y.html` pages to be generated (10 years of data takes a while!).

That explains why those `[[ToDate]]` templates didn't get the month links, but the `[[SummaryByMonth]]` and `[[SummaryByYear]]` templates did.  🤦🏻‍♂️

I put those sections back in the order they were originally, and now every year has a month link on all pages.

**Reply by Daveiano** [reply]
Hi there!

Glad you find the cause, good catch! It seems that weewx needs to process these `SummaryBy` pages first, to get the data for the other templates.
